src/seedu/addressbook/data/person/Address.java

```diff
@@ -6,27 +6,26 @@
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
-public class Address {
+public class Address extends Contact {
 
     public static final String EXAMPLE = "123, some street";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
-    public final String value;
-    private boolean isPrivate;
+    public Contact data = new Contact();
```

Comment:
Woah, this is incorrect. A subclass automatically inherit the members of the parent class, i.e. you dont need a "superclass object" in order to have access to those members.